lib/libc++/Makefile

```diff
@@ -77,6 +82,11 @@ CXXFLAGS+=	-std=c++11
 .endif
 
 LIBADD+=	cxxrt
+.ifdef LIBCHERI
+
+# XXXAR: why is this needed?
+LDFLAGS+=	-L../libcxxrt
```

Comment:
I am pretty sure libcxxrt and libc++ do need to be added to the buildcheri target regardless.